qui/updater/summary_page.py

```diff
@@ -0,0 +1,331 @@
+            "restart_checkbox_header")
+        self.head_checkbox_button.set_inconsistent(True)
+        self.head_checkbox_button.connect(
+            "toggled", self.on_header_toggled)
+        self.head_checkbox = HeaderCheckbox(
+            self.head_checkbox_button,
+            allowed=["", "", "EXCLUDED"],
+            callback_all=lambda plural, num: self.next_button.set_label(
+                f"_Finish and restart {num} qube{plural}"),
+            callback_some=lambda plural, num: self.next_button.set_label(
+                f"_Finish and restart {num} qube{plural}"),
+            callback_none=lambda *_args: self.next_button.set_label("_Finish"),
+        )
+
+        self.summary_list = self.builder.get_object("summary_list")
+        self.summary_list.connect("row-activated", back_by_row_selection)
+
+    @disable_checkboxes
+    def on_checkbox_toggled(self, _emitter, path, *_args):
+        """Handles (un)selection of single row."""
+        if path is None:
+            return
+
+        self.list_store.invert_selection(path)
+        self.refresh_buttons()
+
+    def refresh_buttons(self):
+        """Refresh additional info column and finish button info."""
+        for row in self.list_store:
+            row.refresh_additional_info()
+        selected_num = sum(
+            row.selected for row in self.list_store)
+        if selected_num == 0:
+            self.head_checkbox.state = HeaderCheckbox.NONE
+        elif selected_num == len(self.list_store):
+            self.head_checkbox.state = HeaderCheckbox.ALL
+        else:
+            self.head_checkbox.state = HeaderCheckbox.SELECTED
+        plural = "s" if selected_num > 1 else ""
+        self.head_checkbox.set_buttons(plural, selected_num)
+
+    @disable_checkboxes
+    def on_header_toggled(self, _emitter):
+        """Handles clicking on header checkbox.
+
+        Cycle between selection from appvms which templates was updated :
+         <1> only sys-vms
+         <2> sys-vms + other appvms but without excluded in settings
+         <3> all appvms
+         <4> no vm. (nothing)
+
+        If the user has selected any vms that do not match the defined states,
+        the cycle will start from (1).
+        """
+        on_head_checkbox_toggled(
+            self.list_store, self.head_checkbox, self.select_rows)
+
+    @property
+    def is_populated(self) -> bool:
+        """Returns True if restart list is populated."""
+        return self.list_store is not None
+
+    @property
+    def is_visible(self):
+        """Returns True if page is shown by stack."""
+        return self.stack.get_visible_child() == self.page
+
+    def show(
+            self,
+            qube_updated_num: int,
+            qube_no_updates_num: int,
+            qube_failed_num: int
+    ):
+        """Show this page and handle buttons."""
+        self.stack.set_visible_child(self.page)
+        qube_updated_plural = "s" if qube_updated_num != 1 else ""
```

Comment:
That's not a great way of doing it, because different languages have very different plural rules. (Like Polish) . Is it possible to use ngettext here?